### PR TITLE
MM-19183: handle plugins without homepage

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Locally installed plugins will appear in the plugin marketplace but won't necessary have homepage metadata. When this occurs, don't allow clicking on the row, since the user would just end up at the current URL in a new tab.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-19183

### CODE REVIEW

File: components/plugin_marketplace/marketplace_item/marketplace_item.jsx [hunk 1]
```diff
@@ -157,25 +157,46 @@ export default class MarketplaceItem extends React.Component {
                 pluginIcon = <PluginIcon className='icon__plugin icon__plugin--background'/>;
             }
 
+            let pluginDetails = (
+                <>
+                    {this.props.name} <span className='light subtitle'>{versionLabel}</span>
+                    <p className={'more-modal__description' + (this.state.serverError ? ' error_text' : '')}>
```
Comment (ali-farooq0) [inline components/plugin_marketplace/marketplace_item/marketplace_item.jsx]: Recommend using `classNames` package here
```
<p className={classNames('more-modal__description', {'error_text': this.state.serverError})}>
```

File: components/plugin_marketplace/marketplace_item/marketplace_item.jsx [hunk 2]
```diff
@@ -157,25 +157,46 @@ export default class MarketplaceItem extends React.Component {
                 pluginIcon = <PluginIcon className='icon__plugin icon__plugin--background'/>;
             }
 
+            let pluginDetails = (
+                <>
```
Comment (ali-farooq0) [inline components/plugin_marketplace/marketplace_item/marketplace_item.jsx]: Nice! Didn't know about this shorthand. 👌

File: components/plugin_marketplace/marketplace_item/marketplace_item.jsx [hunk 3]
```diff
@@ -157,25 +157,46 @@ export default class MarketplaceItem extends React.Component {
...
+                    {this.props.name} <span className='light subtitle'>{versionLabel}</span>
+                    <p className={'more-modal__description' + (this.state.serverError ? ' error_text' : '')}>
+                        { this.state.serverError ? this.state.serverError : this.props.description}
```
Comment (ali-farooq0) [inline components/plugin_marketplace/marketplace_item/marketplace_item.jsx]:
```suggestion
                        {this.state.serverError ? this.state.serverError : this.props.description}
```

File: components/plugin_marketplace/marketplace_item/marketplace_item.jsx [hunk 4]
```diff
@@ -157,25 +158,46 @@ export default class MarketplaceItem extends React.Component {
...
+                    >
+                        {pluginDetails}
+                    </span>
+                );
+            }
+
             return (
                 <div
-                    className={'more-modal__row more-modal__row--link' + (this.state.serverError ? ' item_error' : '')}
+                    className={classNames('more-modal__row', 'more-modal__row--link', {item_error: this.state.serverError})}
```
Comment (ali-farooq0) [inline components/plugin_marketplace/marketplace_item/marketplace_item.jsx]: ❤️

**Comment by ali-farooq0** [comment]
@prapti ready to be reviewed.

**Comment by prapti** [comment]
@ali-farooq0 @lieut-data Does the "Expected" on the [ticket](https://mattermost.atlassian.net/browse/MM-19183) need to be updated with something more concrete?

**Comment by lieut-data** [comment]
@prapti, done.

**Pull request merged**